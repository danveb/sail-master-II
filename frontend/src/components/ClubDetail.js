import React, { useState, useEffect } from 'react' 
import SailMasterIIApi from '../API/api'
import { Card } from 'react-bootstrap' 
import { useParams } from 'react-router-dom'

const ClubDetail = () => {
    // useState
    const [club, setClub] = useState([])
    const { id } = useParams()

    // useEffect
    useEffect(() => {
        async function getClub(id) {
            const response = await SailMasterIIApi.getClub(id) 
            setClub(response)
        }
        getClub(id) 
    }, [id])

    return (
        <Card className="mt-4 ">
            <Card.Body>
                <Card.Title>{club.name}</Card.Title>
                <Card.Text>{`${club.address}, ${club.city}, ${club.state} ${club.state}`}</Card.Text>
                <Card.Text>{`${club.url}`}</Card.Text>
                <Card.Text>{`${club.tel}`}</Card.Text>
                <Card.Text>{`Latitude: ${club.lat} | Longitude: ${club.lon}`}</Card.Text>
            </Card.Body>
        </Card>
    )
}

export default ClubDetail