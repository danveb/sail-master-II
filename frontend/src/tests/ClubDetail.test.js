import React from 'react'
import { render } from '@testing-library/react'
import ClubDetail from '../components/ClubDetail'
// smoke test 
it('renders without crashing', () => {
    render(<ClubDetail />) 
})

// snapshot test
it('matches snapshot', () => {
    const { asFragment } = render(<ClubDetail />)
    expect(asFragment()).toMatchSnapshot()
})