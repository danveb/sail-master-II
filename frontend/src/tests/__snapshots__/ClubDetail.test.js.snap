// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mt-4  card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Club: 
      </div>
      <p
        class="card-text"
      >
        Address: undefined, undefined, undefined undefined
      </p>
      <p
        class="card-text"
      >
        Website:
        <a
          href="https://undefined"
          rel="noreferrer noopener"
          target="_blank"
        >
          undefined
        </a>
      </p>
      <p
        class="card-text"
      >
        Tel: undefined
      </p>
      <p
        class="card-text"
      >
        Latitude: undefined | Longitude: undefined
      </p>
    </div>
  </div>
</DocumentFragment>
`;
